packages/nimble-components/src/select/specs/option-groups-hld.md

```diff
@@ -0,0 +1,79 @@
+# Select and Combobox option groups
+
+## Problem Statement
+
+It is not uncommon to want to categorize the options that are present in the dropdown for a `Select`/`Combobox` component. By supporting categories, users are then able to provide options that have the same display value, but under separate categories, helping avoid confusion.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [Interaction Design](./IxD.md#groups)
+-   [Sectioned Combobox (issue 791)](https://github.com/ni/nimble/issues/791)
+-   [Prototype branch Storybook](https://60e89457a987cf003efc0a5b-cvmxlqwglb.chromatic.com/?path=/docs/components-select--docs) ([code](https://github.com/ni/nimble/tree/groupable-select))
+
+## Implementation / Design
+
+The HTML native `select` supports option groups with the native `optgroup` element. Configuring the `select` to use the `optgroup` takes the following form:
```

Comment:
html optgroup does not support nesting. Would we also only support one layer of nesting?

Reply:
I do not see a reason to support nesting, at least initially. While it's easy to envision a use-case that could leverage nesting, I'm willing to use the native APIs as a support guide, until we have calls to alter that decision.

Reply:
Agreed, I'd call it out that it is unsupported and we should _maybe_ consider testing that edge case to make sure we don't behave poorly (I think it would be natural to assume nesting is supported).

Reply:
Moved this to a 'Future Considerations' section.